modules/core/src/lib/composite-layer.js

```diff
@@ -113,16 +113,26 @@ export default class CompositeLayer extends Layer {
         target: []
       };
       return (x, i) => {
-        if (x.__source) {
-          objectInfo.index = x.__source.index;
-          return accessor(x.__source.object, objectInfo);
+        // Iterable data
+        if (x) {
+          if (x.__source) {
+            objectInfo.index = x.__source.index;
+            return accessor(x.__source.object, objectInfo);
+          }
+        } else {
+          // Non-iterable data
+          x = {
+            properties: this.getNonIterableProperties(i)
+          };
         }
         return accessor(x, i);
       };
     }
     return accessor;
   }
 
+  getNonIterableProperties(i) {}
```

Comment:
As discussed, the accessor needs to be wrapped before passing to `GeoJsonLayer`.